homeassistant/components/overkiz/climate_entities/hitachi_air_to_air_heat_pump_ovp.py

```diff
@@ -0,0 +1,357 @@
+
+OVERKIZ_TO_HVAC_MODES: dict[str, HVACMode] = {
+    OverkizCommandParam.AUTOHEATING: HVACMode.AUTO,
+    OverkizCommandParam.AUTOCOOLING: HVACMode.AUTO,
+    OverkizCommandParam.ON: HVACMode.HEAT,
+    OverkizCommandParam.OFF: HVACMode.OFF,
+    OverkizCommandParam.HEATING: HVACMode.HEAT,
+    OverkizCommandParam.FAN: HVACMode.FAN_ONLY,
+    OverkizCommandParam.DEHUMIDIFY: HVACMode.DRY,
+    OverkizCommandParam.COOLING: HVACMode.COOL,
+}
+
+HVAC_MODES_TO_OVERKIZ: dict[HVACMode, str] = {
+    HVACMode.AUTO: OverkizCommandParam.AUTO,
+    HVACMode.HEAT: OverkizCommandParam.HEATING,
+    HVACMode.OFF: OverkizCommandParam.HEATING,
+    HVACMode.FAN_ONLY: OverkizCommandParam.FAN,
+    HVACMode.DRY: OverkizCommandParam.DEHUMIDIFY,
+    HVACMode.COOL: OverkizCommandParam.COOLING,
+}
+
+OVERKIZ_TO_SWING_MODES: dict[str, str] = {
+    OverkizCommandParam.BOTH: SWING_BOTH,
+    OverkizCommandParam.HORIZONTAL: SWING_HORIZONTAL,
+    OverkizCommandParam.STOP: SWING_OFF,
+    OverkizCommandParam.VERTICAL: SWING_VERTICAL,
+}
+
+SWING_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_SWING_MODES.items()}
+
+OVERKIZ_TO_FAN_MODES: dict[str, str] = {
+    OverkizCommandParam.AUTO: FAN_AUTO,
+    OverkizCommandParam.HIGH: FAN_HIGH,  # fallback, state can be exposed as HIGH, new state = hi
+    OverkizCommandParam.HI: FAN_HIGH,
+    OverkizCommandParam.LOW: FAN_LOW,
+    OverkizCommandParam.LO: FAN_LOW,
+    OverkizCommandParam.MEDIUM: FAN_MEDIUM,  # fallback, state can be exposed as MEDIUM, new state = med
+    OverkizCommandParam.MED: FAN_MEDIUM,
+    OverkizCommandParam.SILENT: OverkizCommandParam.SILENT,
+}
+
+FAN_MODES_TO_OVERKIZ: dict[str, str] = {
+    FAN_AUTO: OverkizCommandParam.AUTO,
+    FAN_HIGH: OverkizCommandParam.HI,
+    FAN_LOW: OverkizCommandParam.LO,
+    FAN_MEDIUM: OverkizCommandParam.MED,
+    FAN_SILENT: OverkizCommandParam.SILENT,
+}
+
+
+class HitachiAirToAirHeatPumpOVP(OverkizEntity, ClimateEntity):
+    """Representation of Hitachi Air To Air HeatPump."""
+
+    _attr_hvac_modes = [*HVAC_MODES_TO_OVERKIZ]
+    _attr_fan_modes = [*FAN_MODES_TO_OVERKIZ]
+    _attr_preset_modes = [PRESET_NONE, PRESET_HOLIDAY_MODE]
+    _attr_swing_modes = [*SWING_MODES_TO_OVERKIZ]
+    _attr_target_temperature_step = 1.0
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_translation_key = DOMAIN
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.PRESET_MODE
+        )
+
+        if self.device.states.get(OverkizState.OVP_SWING):
+            self._attr_supported_features |= ClimateEntityFeature.SWING_MODE
+
+        if self._attr_device_info:
+            self._attr_device_info["manufacturer"] = "Hitachi"
```

Comment:
this is not good as it overrides the existing dictionary.

Reply:
I am not sure if I fully understand. We set a default value in the entity.py, where we first check if the device has a specific vendor name, and otherwise fallback to the hub vendor (e.g. Atlantic or Somfy). For this device they don't provide the vendor name in the device information, and falling back to the hub vendor name doesn't make sense. Hence we overwrite the current value with 'Hitachi', which is the vendor of this specific device.

This pattern is used more broadly in Overkiz, so I will make more changes depending on the outcome :).

Reply:
Ah jus checked the other code, looks like it's fine. I thought it was overriding a key in a global dictionary but it's generated in the constructor.